core/src/subgraph/registrar.rs

```diff
@@ -292,88 +294,60 @@ where
-        let chain_stores = self.chain_stores.clone();
-        let ethereum_adapters = self.ethereum_adapters.clone();
-        let version_switching_mode = self.version_switching_mode;
-
+    ) -> DynTryFuture<'a, (), SubgraphRegistrarError> {
         let logger = self.logger_factory.subgraph_logger(&hash);
-        let logger_for_subgraph_version = logger.clone();
-        let logger_for_debug = logger.clone();
-        let name_inner = name.clone();
 
-        Box::new(
-            UnvalidatedSubgraphManifest::resolve(
+        async move {
+            let unvalidated = UnvalidatedSubgraphManifest::resolve(
                 hash.to_ipfs_link(),
                 self.resolver.clone(),
-                logger,
+                &logger,
             )
             .map_err(SubgraphRegistrarError::ResolveError)
-            .and_then(move |unvalidated| {
-                future::result(unvalidated.validate(store_for_validation)).map_err(
-                    |validation_errors| {
-                        SubgraphRegistrarError::ManifestValidationError(validation_errors)
-                    },
-                )
-            })
-            .and_then(move |(manifest, validation_warnings)| {
-                let network_name = manifest.network_name();
-                chain_stores
-                    .clone()
-                    .get(&network_name)
-                    .ok_or(SubgraphRegistrarError::NetworkNotSupported(
-                        network_name.clone(),
-                    ))
-                    .and_then(move |chain_store| {
-                        ethereum_adapters
-                            .get(&network_name)
-                            .ok_or(SubgraphRegistrarError::NetworkNotSupported(
-                                network_name.clone(),
-                            ))
-                            .map(move |ethereum_adapter| {
-                                (
-                                    manifest,
-                                    ethereum_adapter.clone(),
-                                    chain_store.clone(),
-                                    validation_warnings,
-                                )
-                            })
-                    })
-            })
-            .and_then(
-                move |(manifest, ethereum_adapter, chain_store, validation_warnings)| {
-                    let manifest_id = manifest.id.clone();
-                    create_subgraph_version(
-                        &logger_for_subgraph_version,
-                        store_for_subgraph_version,
-                        chain_store.clone(),
-                        ethereum_adapter.clone(),
-                        name,
-                        manifest,
-                        node_id,
-                        version_switching_mode,
-                    )
-                    .map(|_| (manifest_id, validation_warnings))
-                },
+            .await?;
```

Comment:
The indentation (or lack thereof) here looks odd. Was this formatted with `rustfmt`?

Reply:
Yes, it has been formatted.

Reply:
Ok, nevermind then. 😉